src/packagedcode/go_mod.py

```diff
@@ -158,6 +168,67 @@ def parse_gomod(location):
                     )
             continue
 
+        if "replace" in line and "(" in line:
+            for exc in lines[i + 1 :]:
+                exc = preprocess(exc)
+                if ")" in exc:
+                    break
+
+                parsed_rep_link = parse_rep_link(exc)
+                ns_name = parsed_rep_link.group("ns_name")
+                replacement_ns_name = parsed_rep_link.group("replacement_ns_name")
+                namespace, _, name = ns_name.rpartition("/")
+                replacement_namespace, _, replacement_name = (
+                    replacement_ns_name.rpartition("/")
+                )
+
+                replace.append(
+                    GoModule(
+                        namespace=namespace,
+                        name=name,
+                        version=parsed_rep_link.group("version"),
+                    )
+                )
+
+                replace.append(
+                    GoModule(
+                        namespace=replacement_namespace,
+                        name=replacement_name,
+                        version=parsed_rep_link.group("replacement_version"),
+                    )
+                )
+            continue
+
+        if "replace" in line and "=>" in line:
+
+            line = line.lstrip("replace").strip()
+
+            parsed_rep_link = parse_rep_link(line)
+            ns_name = parsed_rep_link.group("ns_name")
+            replacement_ns_name = parsed_rep_link.group("replacement_ns_name")
+            namespace, _, name = ns_name.rpartition("/")
+            replacement_namespace, _, replacement_name = replacement_ns_name.rpartition(
+                "/"
+            )
+
+            replace.append(
+                GoModule(
+                    namespace=namespace,
+                    name=name,
+                    version=parsed_rep_link.group("version"),
+                )
+            )
+
+            replace.append(
+                GoModule(
+                    namespace=replacement_namespace,
+                    name=replacement_name,
+                    version=parsed_rep_link.group("replacement_version"),
+                )
+            )
```

Comment:
This is not correct, as we are treating the module to be replaced and the replacement module the same way. I think the first module in the replace directive (the module to be replaced) should be added as a dependency with scope as `exclude` and the second module in the replace directive (the replacement module) should be added with scope as `require`.

@pombredanne what do you think?

Reply:
We need to track the origin and renamed module separately when parsing alright.
And then we would need to only report the replaced modules and not both... and there could be trailing comments in replace that are also useful to track